### PR TITLE
PMM-1638: Enhanced Monitoring metrics

### PR DESCRIPTION
**Pull request by arvenil**
- [x] Integrate Mykola's rdsosmetrics_exporter into rds_exporter, with changes from https://github.com/Percona-Lab/rdsosmetrics_exporter/pull/1:
	- [x] Scrape CloudWatch only when Prometheus scrapes exporter.
	- [x] Scrape all instances independently, in parallel.
	- [x] Scrape all instance metrics in parallel.
- [x] Use proper metrics types and descriptions.
- [x] Use node_exporter metric names, types, and descriptions when possible.
	- [x] cpuUtilization.idle
	- [x] cpuUtilization.nice
	- [x] cpuUtilization.steal
	- [x] cpuUtilization.wait
	- [x] cpuUtilization.system
	- [x] cpuUtilization.guest
	- [x] cpuUtilization.user
	- [x] cpuUtilization.irq
	- [x] cpuUtilization.total
	- [x] loadAverageMinute.one
	- [x] memory.buffers
	- [x] memory.cached
	- [x] memory.free
	- [x] memory.total
	- [x] swap.free
	- [x] swap.total
	- [x] tasks.blocked
	- [x] tasks.running
	- [x] memory.active
	- [x] memory.inactive
	- [x] memory.slab
	- [x] memory.mapped
	- [x] memory.pageTables
	- [x] diskIO[0].readKb
	- [x] diskIO[0].writeKb
	- [x] fileSys.name
	- [x] fileSys.total
	- [x] memory.dirty
- [x] Add latency metric: the difference between the current time and timestamp in the metric itself.
- [x] Expose enhanced metrics on the separate endpoint from basic metrics so Prometheus can scrape them more often.

### CODE REVIEW

File: enhanced/metrics.go [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+			"The amount of virtual memory allocated to the process, in kilobytes.",
+			[]string{"instance", "region", "id"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_processList_vmlimit": {
+		Name: "vmlimit",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_processList_vmlimit",
+			"TODO",
```
Comment (arvenil) [inline enhanced/metrics.go]: @AlekSi some metrics are missing in http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Monitoring.OS.html

File: enhanced/metrics.go [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+	"rdsosmetrics_memory_hugePagesFree": {
+		Name: "hugePagesFree",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_hugePagesFree",
+			"The number of free huge pages.Huge pages are a feature of the Linux kernel.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_hugePagesRsvd": {
```
Comment (arvenil) [inline enhanced/metrics.go]: @AlekSi What should be name convention for those metrics? For basic metrics we have e.g. `aws_rds_swap_usage_average` should this one be called `aws_rds_memory_hugepagesrsvd` ?

File: enhanced/generate/main.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+			"util":            "The percentage of CPU time during which requests were issued. This metric is not available for Amazon Aurora.",
+			"writeIOsPS":      "The number of write operations per second. This metric is not available for Amazon Aurora.",
+			"writeKb":         "The total number of kilobytes written. This metric is not available for Amazon Aurora.",
+			"writeKbPS":       "The number of kilobytes written per second. This metric is not available for Amazon Aurora.",
+			"wrqmPS":          "The number of merged write requests queued per second. This metric is not available for Amazon Aurora.",
+			"readLatency":     "TODO",
+			"writeLatency":    "TODO",
+			"writeThroughput": "TODO",
+			"readThroughput":  "TODO",
+			"diskQueueDepth":  "TODO",
```
Comment (arvenil) [inline enhanced/generate/main.go]: @AlekSi above metrics in vlimit are missing in http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Monitoring.OS.html

File: enhanced/metrics.go (hunk 2 above)
Comment (AlekSi) [reply enhanced/metrics.go]: @delgod Does AWS has some advice for non-documented metrics? Are they stable? Can they be _suddenly_ removed?

Maybe we should just skip them?

File: enhanced/generate/main.go (hunk 3 above)
Comment (AlekSi) [reply enhanced/generate/main.go]: @delgod Does AWS has some advice for non-documented metrics? Are they stable? Can they be _suddenly_ removed?

Maybe we should just skip them?​

File: enhanced/metrics.go (hunk 2 above)
Comment (AlekSi) [reply enhanced/metrics.go]: Is it the same as `node_memory_HugePages_Rsvd`? If yes, then we should use that name.

Comment (arvenil) [reply enhanced/metrics.go]: Hmm, was not in https://confluence.percona.com/display/PMM/AWS%3A+RDS+Metrics+in+node_exporter+format

@delgod Maybe depends on env/os?

File: enhanced/generate/main.go (hunk 3 above)
Comment (delgod) [reply enhanced/generate/main.go]: I think we can scrape them, but shouldn't rely on.

File: enhanced/metrics.go (hunk 2 above)
Comment (delgod) [reply enhanced/metrics.go]: oh, I haven't seen `node_memory_HugePages_Rsvd` metrics on CentOS, going to change our wiki

**Pull request merged**